core/chains/evm/chain.go

```diff
@@ -104,7 +104,7 @@ func newChain(dbchain types.Chain, opts ChainSetOpts) (*chain, error) {
 		headSaver = headtracker.NewHeadSaver(headTrackerLogger, orm, cfg)
 		headTracker = headtracker.NewHeadTracker(headTrackerLogger, client, cfg, headBroadcaster, headSaver)
 	} else {
-		headTracker = opts.GenHeadTracker(dbchain)
+		headTracker = opts.GenHeadTracker(dbchain, headBroadcaster)
```

Comment:
if GenHeadTracker wants to construct a head tracker with NewHeadTracker, it needs the headBroadcaster which is started when the chain is started or the head tracker will not work